### PR TITLE
feat: add uk stores

### PR DESCRIPTION
**Pull request by andrewmackrodt**
**Relates to #461 - merge second**

This adds 10 UK stores, I've not added `maxPrice` yet but plan to do so in a future PR. https://github.com/andrewmackrodt/nvidia-snatcher/compare/pr-framework-updates...pr-uk-stores

### CODE REVIEW

File: src/store/model/helpers/card.ts [hunk 1]
```diff
@@ -1,13 +1,91 @@
...
+	};
+	/* eslint-enable unicorn/no-fn-reference-in-iterator */
+}
+
 export function parseCard(name: string): Card | null {
-	name = name.replace(/[^\w ]+/g, '').trim();
-	name = name.replace(/\bgraphics card\b/gi, '').trim();
-	name = name.replace(/\b\w+ fan\b/gi, '').trim();
-	name = name.replace(/\s{2,}/g, ' ');
+	name = name.replace(/\w+-\w+-[^ ]+/g, '');
```
Comment (jef) [inline src/store/model/helpers/card.ts:75]: Wow 😮 Nice work!

File: src/store/model/amazon-uk.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	links: [
+		{
+			brand: 'test:brand',
+			cartUrl: 'https://www.amazon.co.uk/gp/aws/cart/add.html?ASIN.1=B081265T5Z&Quantity.1=1',
+			model: 'test:model',
+			series: 'test:series',
+			url: 'https://www.amazon.co.uk/dp/B081265T5Z/'
+		}
+	],
+	linksBuilder: {
```
Comment (jef) [inline src/store/model/amazon-uk.ts:39]: I imagine we could use this for the other Amazon stores, but could be very wrong there.

**Review by jef — APPROVED** [review]
Wowa wowa. It looks great. Thanks for putting in this work. You're on the hunt for that card 😎

**Comment by jef** [comment]
> We can break this into 2 different PRs or commits squashed into the 2 units once approved.

No problem. I don't mind it in both.

**Comment by andrewmackrodt** [comment]
> You're on the hunt for that card sunglasses

I actually snagged a 3080 the same night/morning nvidia made the API changes but trying to do what little I can to help others get their cards :)

**Comment by jef** [comment]
Well we all appreciate your contribution ❤️

File: src/store/model/amazon-uk.ts (hunk 2 above)
Comment (andrewmackrodt) [reply src/store/model/amazon-uk.ts:39]: I imagine the `builder` function would work between stores, but determining the `urls[].url[]` would require a manual map per region. At a minimum, the following may work (not tested it compiles let alone works):

```ts
function getAmazonQueryString(search: string, categoryId: number, includeOutOfStockId: number) {
    return [
        `k=${encodeURIComponent(search)}`,
        'rh=' + encodeURIComponent([
            `n:${categoryId}`,
            `p_n_availability:${includeOutOfStockId}`
        ].join(','))
    ].join('&');
}

const queryString = getAmazonQueryString('+"RTX 3080" -2080 -GTX', 430500031, 419162031);
const domain = 'www.amazon.co.uk';
const url = `https://${domain}/s?${queryString}`;
```

**Comment by andrewmackrodt** [comment]
@jef I've split the first changset into #461 and will rebase this PR on top of main once that one's merged.

File: src/store/model/amazon-uk.ts (hunk 2 above)
Comment (jef) [reply src/store/model/amazon-uk.ts:39]: That makes sense to me. Maybe we can bring this in at a later point. Thanks for checking this out.

**Review by jef — APPROVED** [review]
Thank you very much for all this hard work!

**Pull request merged**